lib/resources/auditd.rb

```diff
@@ -0,0 +1,236 @@
+# encoding: utf-8
+# copyright: 2015, Vulcano Security GmbH
+# author: Christoph Hartmann
+# author: Dominik Richter
+# author: Jen Burns
+
+require 'forwardable'
+require 'utils/filter_array'
+require 'utils/filter'
+require 'utils/parser'
+
+module Inspec::Resources
+  # rubocop:disable Metrics/ClassLength
+  class AuditDaemon < Inspec.resource(1)
+    extend Forwardable
+    attr_accessor :lines
+    attr_reader :params
+
+    name 'auditd'
+    desc 'Use the auditd InSpec audit resource to test the rules for logging that exist on the system. The audit.rules file is typically located under /etc/audit/ and contains the list of rules that define what is captured in log files. These rules are output using the auditcl -l command.'
+    example "
+      describe auditd.syscall('chown').where {arch == 'b32'} do
+        its('action') { should eq ['always'] }
+        its('list') { should eq ['exit'] }
+      end
+
+      describe auditd.where {key == 'privileged'} do
+        its('permissions') { should include ['x'] }
+      end
+
+      describe auditd do
+        its('lines') { should include %r(-w /etc/ssh/sshd_config) }
+      end
+    "
+
+    def initialize
+      @content = inspec.command('/sbin/auditctl -l').stdout.chomp
+      @params = []
+
+      if @content =~ /^LIST_RULES:/
+        return skip_resource 'The version of audit is outdated. The `auditd` resource supports versions of audit >= 2.3.'
+      else
+        parse_content
+        @legacy = nil
+      end
+    end
+
+    filter = FilterTable.create
+    filter.add_accessor(:where)
+          .add_accessor(:entries)
+          .add(:file,         field: 'file')
+          .add(:list,         field: 'list')
+          .add(:action,       field: 'action')
+          .add(:fields,       field: 'fields')
+          .add(:fields_nokey, field: 'fields_nokey')
+          .add(:syscall,      field: 'syscall')
+          .add(:key,          field: 'key')
+          .add(:arch,         field: 'arch')
+          .add(:path,         field: 'path')
+          .add(:permissions,  field: 'permissions')
+          .add(:exit,         field: 'exit')
+
+    filter.connect(self, :params)
+
+    def status(name = nil)
+      return @legacy.status(name) if @legacy
+
+      @status_content ||= inspec.command('/sbin/auditctl -s').stdout.chomp
+      @status_params ||= Hash[@status_content.scan(/^([^ ]+) (.*)$/)]
+
+      return @status_params[name] if name
+      @status_params
+    end
+
+    def parse_content
+      @lines = @content.lines.map(&:chomp)
+
+      lines.each do |line|
+        if is_file_syscall_syntax?(line)
+          get_file_syscall_syntax_rules(line)
+        end
+
+        if is_syscall?(line)
+          get_syscall_rules(line)
+
+        elsif is_file?(line)
+          get_file_rules(line)
+        end
+      end
+    end
+
+    def get_file_syscall_syntax_rules(line)
+      file = get_file_syscall_syntax line
```

Comment:
Sorry @jburns12 if I wasn't clear. The standard is to *use* parens rather than remove them, so can you update these lines as such?

```
file = get_file_syscall_syntax(line)
action, list = get_action_list(line)
... etc ...
```

... and throughout the resource, please?  Thank you! 🙂